.github/workflows/validate.yml

```diff
@@ -57,35 +57,21 @@ jobs:
 
       - uses: actions/checkout@v3
 
-      # See https://github.com/haskell/cabal/pull/8739
-      - name: Sudo chmod to permit ghcup to update its cache
-        run: |
-          if [[ "${{ runner.os }}" == "Linux" ]]; then
-            sudo ls -lah /usr/local/.ghcup/cache
-            sudo mkdir -p /usr/local/.ghcup/cache
-            sudo ls -lah /usr/local/.ghcup/cache
-            sudo chown -R $USER /usr/local/.ghcup
-            sudo chmod -R 777 /usr/local/.ghcup
-          fi
+      - uses: actions/cache@v3
+        with:
+          # validate.sh uses a special build dir
+          path: |
+            ${{ steps.setup-haskell.outputs.cabal-store }}
+            dist-*
+          key: ${{ runner.os }}-${{ matrix.ghc }}-${{ github.sha }}
+          restore-keys: ${{ runner.os }}-${{ matrix.ghc }}-
+
       - uses: haskell/actions/setup@v2
         id: setup-haskell
         with:
           ghc-version: ${{ matrix.ghc }}
           cabal-version: '3.8.1.0'
```

Comment:
Let's see what CI says. I'm curious if it will twist our hand to bump the cabal version here or not.

Reply:
(My bad, i always rush to hit Comment...) My worry was specifically about the test suite : it's very peculiar about the version of cabal-install you're compiling it with. It now fails with 3.10. You probably will have to try to revert to 3.8 and see if that works...